packages/blaze/template.js

```diff
@@ -56,6 +62,52 @@ Blaze.isTemplate = function (t) {
   return (t instanceof Blaze.Template);
 };
 
+/**
+ * @name  onCreated
+ * @instance
+ * @memberOf Template
+ * @summary Add a callback when an instance of a template is created.
+ * @param {Function} callback A function to be added as a callback.
+ * @locus Client
+ */
+Template.prototype.onCreated = function (cb) {
+  this._callbacks.created.push(cb);
+};
+
+/**
+ * @name  onRendered
+ * @instance
+ * @memberOf Template
+ * @summary Add a callback when an instance of a template is rendered.
+ * @param {Function} callback A function to be added as a callback.
+ * @locus Client
+ */
+Template.prototype.onRendered = function (cb) {
+  this._callbacks.rendered.push(cb);
+};
+
+/**
+ * @name  onDestroyed
+ * @instance
+ * @memberOf Template
+ * @summary Add a callback when an instance of a template is destroyed.
```

Comment:
I think mentioning the magical "destroyed" helps, although Blaze can remove a tree from DOM and then attach it somewhere else.